runtime-modules/content/src/lib.rs

```diff
@@ -3195,6 +3234,12 @@ impl<T: Config> Module<T> {
             channel.num_videos = channel.num_videos.saturating_sub(1)
         });
 
+        // Repay video state bloat bond
+        let channel_account = ContentTreasury::<T>::account_for_channel(channel_id);
+        video
+            .video_state_bloat_bond
+            .repay::<T>(&channel_account, false)?;
```

Comment:
> a last line of defense against bugs

That's how I understand it in this case. If we make this call infallible we're essentially resigning from this last line of defense. This call is never supposed to fail, because the channel account should always have enough funds to repay the bloat bond, but just in case there is some unexpected state I think it's a good idea to keep this call fallible.

Reply:
Ok, I think I am wrong here, in the context of transactional rollbacks this is actually quite valuable in the real word, I think I just now am worried about the inconsistency, as we have not followed this rule so far, but better late than never to start this convention.

I guess my question is then: is there any way we can make it super obvious to the reader that this call should not fail, its qualitatively different to the calls before the mutation safe barrier? I thinks this is easy to miss? What about something like

```Rust
let paid_channel_bloat_bond_result = <thing that should not fail>;

// Cannot fail
debug_assert!(paid_channel_bloat_bond_result.is_ok());

let channel_account = paid_channel_bloat_bond_result?
```

It's a bit verbose, but feels like it really reminds the reader that this is different, is loud when somethign happens in debug mode, and enjoys the rollback last line of defence in production. Could probably be made very pretty with some wrapper macro which can be used to do this pattern over and over again, like `do_infallible_call!()`? so it is just a one liner.

Reply:
I think this is a good suggestion. There are cases where we might also want to assert that the result of a call is what we expect. Example with `balances::slash()` here https://github.com/Joystream/joystream/blob/carthage/runtime-modules/membership/src/lib.rs#L881

And to catch these debug assertions, we should probably change our CI checks to not use `--release` flag when running the cargo unit tests - https://github.com/Joystream/joystream/blob/carthage/scripts/cargo-checks.sh#L23